ssl/ssl_init.c

```diff
@@ -129,3 +128,46 @@ int OPENSSL_init_ssl(uint64_t opts, const OPENSSL_INIT_SETTINGS *settings)
 
     return 1;
 }
+
+#if defined(__GNUC__) && !defined(_AIX)
+
+void ssl_call_stop(void);
+
+void __attribute__((destructor))
+ssl_call_stop(void)
+{
+    ssl_library_stop();
+}
+
+#elif (defined(__sun) || defined(_AIX)) && !defined(__GNUC__)
+void ssl_call_stop(void);
+# pragma fini(ssl_call_stop)
+void ssl_call_stop(void)
+{
+    ssl_library_stop();
+}
+
+#elif defined(__hpux)
+void ssl_call_stop(void);
+# pragma fini "ssl_call_stop"
+void ssl_call_stop(void)
+{
+    ssl_library_stop();
+}
+#elif defined(__TANDEM)
+/* Method automatically called by the NonStop OS prior to unloading the DLL */
+void __TERM__cleanup(void)
+{
+    ssl_library_stop();
+}
+#elif defined(WIN32)
+void ssl_call_stop(void)
+{
+    ssl_library_stop();
+}
+#else
+void _fini(void)
+{
+    ssl_library_stop();
+}
+#endif
```

Comment:
I dislike all of the copy&paste code here. We will inevitably have to tweak the preprocessor stuff here as we make changes for different platforms. We're now going to have 3 different places in the code to remember to keep in alignment. If we kept the OPENSSL_atexit() handler code we wouldn't have to do this.